workspaces/adventure-pack/goodies/typescript/Iterator.from/index.ts

```diff
@@ -0,0 +1,44 @@
+import { iteratorPrototype } from "../Iterator.prototype";
+import "../Iterator.prototype.toIterable";
+
+declare global {
+  interface IteratorConstructor {
+    from<T>(
+      object: Iterator<T> | Iterable<T> | { next(): IteratorResult<T> },
+    ): IterableIterator<T>;
+  }
+
+  const Iterator: IteratorConstructor;
+}
+
+(globalThis as Record<string, unknown>).Iterator ??= {};
+
+(globalThis as Record<string, unknown>).Iterator = {
+  from<T>(
+    object: Iterator<T> | Iterable<T> | { next(): IteratorResult<T> },
+  ): IterableIterator<T> {
+    if (typeof (object as Iterable<T>)[Symbol.iterator] === "function") {
+      const iterable = object as Iterable<T>;
+      const iterator = iterable[Symbol.iterator]();
+      return iteratorPrototype.toIterable.call(iterator) as IterableIterator<T>;
+    }
+
+    if (Object.prototype.isPrototypeOf.call(iteratorPrototype, object)) {
+      return iteratorPrototype.toIterable.call(
+        object as Iterator<T>,
+      ) as IterableIterator<T>;
+    }
```

Comment:
```suggestion
    if (iteratorPrototype.isPrototypeOf(object)) {
      return toIterable.call(object as Iterator<T>);
    }
```

Reply:
I had to do this to suppress the linter

![5c83bf9be54228f5f5663182b8b717d1](https://github.com/user-attachments/assets/d339d373-ee0f-4265-a967-7d7a36a5ae22)